### PR TITLE
[RHCLOUD-21159] feature: update source types for the provisioning application

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
The provisioning team is working on making the application compatible with Azure and Google Cloud. They need new authentication types for those clouds and the ability to create sources of that type.

## Links

[[RHCLOUD-21159]](https://issues.redhat.com/browse/RHCLOUD-21159)

### CODE REVIEW

File: dao/seeds/source_types.yml [hunk 1]
```diff
@@ -114,6 +114,24 @@ google:
         isRequired: true
         validate:
         - type: required
+    - type: provisioning_project_id_service_account_json
+      name: Provisioning — project ID and Service Account JSON
```
Comment (adiabramovitch) [inline dao/seeds/source_types.yml]: I guess this should be only project id

Comment (MikelAlejoBR) [reply dao/seeds/source_types.yml]: Ooops, you're right. Thanks!

File: dao/seeds/source_types.yml [hunk 2]
```diff
@@ -143,6 +154,20 @@ azure:
...
+          name: authentication.authtype
+          hideField: true
+          initializeOnMount: true
+          initialValue: provisioning_lighthouse_subscription_id
+        - component: text-field
+          name: authentication.username
+          label: Subscription ID
+          isRequired: true
+          validate:
+            - type: required
```
Comment (lzap) [inline dao/seeds/source_types.yml:170]: Can confirm this works fine on our end, assuming there are no other changes from the existing type other than the type/name field :-)

**Review by lindgrenj6 — APPROVED** [review]

**Review by lpichler — APPROVED** [review]

**Comment by lzap** [comment]
We forgot about OpenAPI enums, here is a follow-up PR: https://github.com/RedHatInsights/sources-api-go/pull/566

**Pull request merged**